heudiconv/tests/test_archives.py

```diff
@@ -0,0 +1,90 @@
+from glob import glob
+import os
+import os.path as op
+from pathlib import Path
+import stat
+import typing
+
+
+import pytest
+import shutil
+from .utils import TESTS_DATA_PATH
+from ..parser import get_extracted_dicoms
+
+
+@pytest.fixture
+def get_dicoms_gztar(tmpdir: Path) -> typing.List[str]:
+    tmp_file = tmpdir / "dicom"
+    archive = shutil.make_archive(
+        str(tmp_file),
+        format="gztar", 
+        root_dir=TESTS_DATA_PATH, 
+        base_dir="01-anat-scout")
+    return [archive]
+
+
+@pytest.fixture
+def get_dicoms_zip(tmpdir: Path) -> typing.List[str]:
+    tmp_file = tmpdir / "dicom"
+    archive = shutil.make_archive(
+        str(tmp_file),
+        format="zip", 
+        root_dir=TESTS_DATA_PATH, 
+        base_dir="01-anat-scout")
+    return [archive]
+
+
+@pytest.fixture
+def get_dicoms_list() -> typing.List[str]:
+    return glob(op.join(TESTS_DATA_PATH, "01-anat-scout", "*"))
+
+
+def test_get_extracted_dicoms_single_session_is_none(get_dicoms_gztar: typing.List[str]):
+    for session_, _ in get_extracted_dicoms(get_dicoms_gztar):
+        assert session_ is None
+
+
+def test_get_extracted_dicoms_multple_session_integers(get_dicoms_gztar: typing.List[str]):
+    sessions = []
+    for session, _ in get_extracted_dicoms(get_dicoms_gztar + get_dicoms_gztar):
+        sessions.append(session)
```

Comment:
IMHO 

```suggestion
    sessions = [
        session
        for session, _ in get_extracted_dicoms(get_dicoms_gztar + get_dicoms_gztar)
    ]
```
is a bit cleaner